mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +325,61 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last and 'last_ckpt' in runner.message_hub.runtime_info:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            self._publish_model(runner, last_ckpt)
+
+        if getattr(self, 'best_ckpt_path', None) is not None:
+            self._publish_model(runner, str(self.best_ckpt_path))
+        if getattr(self, 'best_ckpt_path_dict', None) is not None:
+            for key, best_ckpt in self.best_ckpt_path_dict.items():
+                self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, ckpt_path: str) -> None:
+        """Remove unnecessary keys from ckpt_path and save the new checkpoint.
+
+        Args:
+            runner (Runner): The runner of the training process.
+            ckpt_path (str): The checkpoint path that ought to be published.
+        """
+        from mmengine.runner import save_checkpoint
+        from mmengine.runner.checkpoint import _load_checkpoint
+        checkpoint = _load_checkpoint(ckpt_path)
+        published_keys = self.published_keys
```

Comment:
Hi, you can pre-define the type of published_keys to pass mypy check
```suggestion
        published_keys: Union[list, str] = self.published_keys
```